statics/js/components/topology.js

```diff
@@ -263,25 +264,26 @@ var TopologyComponent = {
       this.layout.zoomFit();
     },
 
-    toggleCollapse: function() {
-      this.collapsed = !this.collapsed;
-      this.layout.collapse(this.collapsed);
-    },
-
     collapse: function() {
       this.collapsed = true;
       this.layout.collapse(this.collapsed);
     },
 
-    expand: function() {
-      this.collapsed = false;
-      this.layout.collapse(this.collapsed);
-    },
-
     toggleExpandAll: function(node) {
       this.layout.toggleExpandAll(node);
     },
 
+    expandNextLevel: function() {
+	this.layout.collapseByLevel(this.layout.collapseLevel, !true, this.layout.groups);
+        this.layout.collapseLevel++;
```

Comment:
Feels to me that this.layout.collapseLevel and this.layout.groups should be handled internally by this.layout,
here we should just need to call this.layout.collapseByLevel(true || false)

Reply:
I agree with @eonpatapon that the collapseLevel should be part of the layout. maybe you could add two function one for increase the other for decrease in the layout.

Reply:
done